### PR TITLE
Some simplification to eventful

### PR DESCRIPTION
**Pull request by warmist**
Fixed eventful and other places where it was broken.

### CODE REVIEW

**Comment by lethosor** [comment]
This has some conflicts. Also, can you rebase and update the xml submodule somehow? I think the produce() issues are fixed on develop now.

File: plugins/eventful.cpp [hunk 1]
```diff
@@ -306,11 +287,11 @@ IMPLEMENT_VMETHOD_INTERPOSE(furnace_hook, fillSidebarMenu);
 
 struct product_hook : item_product {
     typedef item_product interpose_base;
-
+
```
Comment (lethosor) [inline plugins/eventful.cpp]: whitespace

File: plugins/eventful.cpp [hunk 2]
```diff
@@ -306,11 +287,11 @@ IMPLEMENT_VMETHOD_INTERPOSE(furnace_hook, fillSidebarMenu);
 
 struct product_hook : item_product {
     typedef item_product interpose_base;
-
+    
     DEFINE_VMETHOD_INTERPOSE(
         void, produce,
         (df::unit *unit,
-         std::vector<void*> *unk,
+         std::vector<df::reaction_product*>* out_products,
```
Comment (lethosor) [inline plugins/eventful.cpp]: Would be nice to match whitespace style here (and below)

**Comment by warmist** [comment]
Oh this is still unmerged? I'll look into it.

**Comment by warmist** [comment]
This is no longer about the vmethod change but about general eventful simplification.

**Pull request merged**